New: Hook to allow inserting custom product head

mrsof opened the pull request:
# New Hook to allow inserting custom product head

A module can insert his own object lines but it can't modify column headers other than via javascript.
Now a module can totally redesign table for object lines.

mrsof committed:
New: Hook to allow inserting custom product head

A module can insert his own object line but it can't modify column headers other than via javascript.
Now a module can totally redesign object line.

mrsof committed:
Fix: update informations

eldy commented on htdocs/core/class/commonobject.class.php:
```diff
@@ -3286,73 +3287,78 @@ function printObjectLines($action, $seller, $buyer, $selected=0, $dateSelector=0
-		if (! empty($conf->global->MAIN_VIEW_LINE_NUMBER)) print '<td class="linecolnum" align="center" width="5">&nbsp;</td>';
+		$parameters = array();
+		$reshook = $hookmanager->executeHooks('printObjectHeads', $parameters, $this, $action); // Note that $action and $object may have been modified by some hooks
+		if (empty($hookmanager->resPrint))
```
Hook must be "replace" hook. It means we must test the value of $reshook to know if we execute standard code or not.

$reshook = $hookmanager->executeHooks('printObjectHeads', $parameters, $this, $action);
print $hookmanager->resPrint;
If (empty($reshook)) 
{
    ...
}

if hook return 0, we output hook + standard code
if hook  return 1, only output of hook is done.

mrsof committed:
New: Hook to allow inserting custom product head

A module can insert his own object lines but it can't modify column headers other than via javascript.
Now a module can totally redesign table for object lines.

mrsof commented:
Thanks.
Changes done as suggested.

eldy commented:
Patch was a little bit modified (hook function renamed) and parameters added for same result. For 6.0
